ktor-server/ktor-server-core/jvm/src/io/ktor/sessions/SessionTransportTransformerEncrypt.kt

```diff
@@ -29,14 +29,18 @@ import javax.crypto.spec.*
 class SessionTransportTransformerEncrypt(
     val encryptionKeySpec: SecretKeySpec,
     val signKeySpec: SecretKeySpec,
-    val ivGenerator: (size: Int) -> ByteArray = { size -> SecureRandom().generateSeed(size) },
+    ivGenerator: ((size: Int) -> ByteArray)? = null,
     val encryptAlgorithm: String = encryptionKeySpec.algorithm,
     val signAlgorithm: String = signKeySpec.algorithm
 ) : SessionTransportTransformer {
     companion object {
         private val log = LoggerFactory.getLogger(SessionTransportTransformerEncrypt::class.qualifiedName)
     }
 
+    private val random by lazy { SecureRandom.getInstance("SHA1PRNG").apply { generateSeed(16) } }
```

Comment:
Is it concurrent safe? If not then it can't be used like this. Perhaps, an object pool is required here.

Reply:
> Instances of java.util.Random are threadsafe. However, the concurrent use of the same java.util.Random instance across threads may encounter contention and consequent poor performance. Consider instead using ThreadLocalRandom in multithreaded designs.
> \- https://docs.oracle.com/javase/7/docs/api/java/util/Random.html

However, using `ThreadLocalRandom` is not cryptographically secure, so don't use that.